claim-check/code-samples/sample-4/azure-function/AzFuncConsumer.cs

```diff
@@ -24,27 +24,19 @@ public static async Task RunAsync([EventHubTrigger("%EVENTHUB_NAME%", Connection
 
             // Downloading the message from blob storage for processing
             string storageConnectionString = Environment.GetEnvironmentVariable("STORAGE_CONNECTION_STRING");
-            CloudStorageAccount storageAccount = null;
-            CloudBlobContainer cloudBlobContainer = null;
 
-            // Check whether the connection string can be parsed.
-            if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
-            {
                 try
                 {
-                    // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
-                    CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                    BlobContainerClient container = new BlobContainerClient(storageConnectionString, containerName);
 
-                    // Get reference to the container
-                    cloudBlobContainer = cloudBlobClient.GetContainerReference(containerName);
-
-                    // Get reference to the blob with heavy payload
-                    CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(blobName);
+                    // Get a reference to a blob named "sample-file" in a container named "sample-container"
+                    BlobClient blob = container.GetBlobClient(blobName);
 
                     log.LogInformation("Here is the large payload information");
 
-                    // Reading payload
-                    Stream blobStream = await cloudBlockBlob.OpenReadAsync();
+                // Reading payload
+                    Stream blobStream=null;
+                    await blob.DownloadToAsync(blobStream);
```

Comment:
This code seems wrong.  I do not believe that copying to a null stream is going to work in the way that you expect.  I think instead you could call `DownloadAsync`, which will give you a `Response<BlobDownloadInfo>`.  The `BlobDownloadInfo.Content` member which you can pass to `new StreamReader()`